pages/api/system/reload.js

```diff
@@ -195,24 +195,64 @@ export default async function handler(req, res) {
       }
 
       // - events
+      async function getCoordinates(city,state,country) {
```

Comment:
After debugging, the following handles the current user input:
```js
async function getCoordinates(city,state,country) {
  const provided = [city, state, country].filter(x => x).join(',');
  if (locationDb[provided]) {
    return locationDb[provided];
  }
  try {
    const location = await fetch(
      `https://nominatim.openstreetmap.org/?addressdetails=1&q=
        ${encodeURIComponent(provided)}&format=json&limit=1`
    );
    const coordinates = await location.json();
    if (coordinates) {
      const point = {
        lat: coordinates[0].lat,
        lon: coordinates[0].lon,
      };
      locationDb[provided] = point;
      return point;
    }
  } catch(e) {
    return null;
  }
  return null;
}
```

`let locationDb = {}` needs to be set at the top of the handler